src/main/java/application/FormController.java

```diff
@@ -341,7 +349,9 @@ public FormEntryResponseBean getNext(@RequestBody SessionRequestBean requestBean
                     formSession.getFormEntryModel(),
                     formSession.getCurrentIndex());
             updateSession(formSession, serializableFormSession);
-            FormEntryResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryResponseBean.class);
+            FormEntryNavigationResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryNavigationResponseBean.class);
+            responseBean.setIsAtLastIndex(formSession.getIsAtLastIndex());
```

Comment:
Right but getCurrentJson does instantiate the object through the indexFromString method. Is also gettable through the getFormIndex() method of FormEntryModel. Writing that though I'm realizing we should definitely unify the way we're passing around the formIndex, though that's maybe out of scope for this PR.